### PR TITLE
Onboard .NET Docker Bot to secret manager

### PR DESCRIPTION
**Pull request by ellahathaway**
Copy from https://github.com/dotnet/dotnet-docker/pull/5797 in dotnet-docker.

Closes https://github.com/dotnet/dotnet-docker/issues/4517

This PR onboards .NET Docker Bot to secret manager and introduces a new pipeline for rotating secrets on a weekly basis.

[Example pipeline run](https://dev.azure.com/dnceng/internal/_build/results?buildId=2519194&view=logs&j=adcf571c-cc13-5e72-b2ad-561cfc37a926)

### CODE REVIEW

**Comment by ellahathaway** [comment]
New pipeline succeeded! 🥳

**Comment by ellahathaway** [comment]
Filed https://github.com/dotnet/dnceng/issues/3842 for some of the issues I encountered while configuring the changes in this PR

File: eng/pipelines/secret-management-weekly.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    steps:
+    - task: UseDotNet@2
+      displayName: Install .NET 8.0 SDK
+      inputs:
+        packageType: sdk
+        version: 8.0.x
+        installationPath: '$(Build.Repository.LocalPath)/.dotnet'
+
+    - task: UseDotNet@2
+      displayName: Install .NET 6.0 runtime
```
Comment (mthalman) [inline eng/pipelines/secret-management-weekly.yml:29]: Does the tool really require .NET 6 runtime?

Comment (ellahathaway) [reply eng/pipelines/secret-management-weekly.yml:29]: Yep - confirmed by pulling [the latest version of the package from the public feed](https://dev.azure.com/dnceng/public/_artifacts/feed/dotnet-eng/NuGet/Microsoft.DncEng.SecretManager/overview/1.1.0-beta.24420.1) and looking at the dll which contained the following lines:

```
// Assembly references:
// System.Runtime, Version=6.0.0.0, Culture=neutral
```

Also did a test run with 8.0 runtime which failed: https://dev.azure.com/dnceng/internal/_build/results?buildId=2521098&view=logs&j=df0a3355-897e-53ae-288d-2f16ccab2822&t=07261306-a000-50cf-efd2-5d677d89250f (internal Microsoft link)

**Review by mthalman — APPROVED** [review]

**Pull request merged**